app/src/main/java/com/github/se/polyfit/model/user/User.kt

```diff
@@ -2,30 +2,34 @@ package com.github.se.polyfit.model.data
 
 import android.net.Uri
 import android.util.Log
+import java.time.LocalDate
 
 data class User(
     var id: String = "",
     var displayName: String? = null,
     var familyName: String? = null,
     var givenName: String? = null,
     var email: String = "",
-    var photoURL: Uri? = null
+    var photoURL: Uri? = null,
+    var heightCm: Long? = null,
+    var weightKg: Double? = null,
+    var dob: LocalDate? = null,
+    var calorieGoal: Long = 2200, // Not sure what we should have this default to
+    var isVegan: Boolean = false,
+    var isVegetarian: Boolean = false,
+    var isGlutenFree: Boolean = false,
+    var isLactoseFree: Boolean = false,
 ) {
 
   fun signOut() {
-    id = ""
-    displayName = null
-    familyName = null
-    givenName = null
-    email = ""
-    photoURL = null
+    update(User()) // Creates an empty user and then updates the current user with it
   }
 
   fun isSignedIn(): Boolean {
     return id.isNotEmpty()
   }
 
-  fun serialize(): Map<String, Any> {
+  fun serialize(): Map<String, Any?> {
```

Comment:
Since not everything has a value, if it's null it doesn't get stored in firebase and then will get reinitialized as null upon deserialiazation through `as?` method